staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go

```diff
@@ -81,6 +81,14 @@ func DialURL(ctx context.Context, url *url.URL, transport http.RoundTripper) (ne
 				tlsConfigCopy.ServerName = inferredHost
 				tlsConfig = tlsConfigCopy
 			}
+
+			// Since this method is primary used within a "Connection: Upgrade" call we assume the caller is
+			// going to write HTTP/1.1 request to the wire. http2 should not be allowed in the TLSConfig.NextProtos,
+			// so we explicitly set that here. We only do this check if the TLSConfig support http/1.1.
+			if supportsHTTP11(tlsConfig.NextProtos) {
+				tlsConfig.NextProtos = []string{"http/1.1"}
```

Comment:
The above block tries hard not to modify the tlsConfig that is passed in. Do we need to make a copy here?

Reply:
You're correct, I thought it was already copied or new at this point but I see I'm wrong.